packages/iov-bns/src/bnsconnection.ts

```diff
@@ -354,9 +353,58 @@ export class BnsConnection implements BcpAtomicSwapConnection {
    * and then continuing with live feeds
    */
   public liveTx(txQuery: BcpTxQuery): Stream<ConfirmedTransaction> {
-    const history = streamPromise(this.searchTx(txQuery));
-    const updates = this.listenTx(txQuery.tags);
-    return Stream.merge(history, updates);
+    let updatesSubscription: Subscription | undefined;
+
+    const producer: Producer<ConfirmedTransaction> = {
+      start: listener => {
```

Comment:
Not easy to deduplicate because we need an object equality operation and JS has no standardized way to do that. But possible by passing a custom equality function.